halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/v1/HalconfigParser.java

```diff
@@ -103,14 +107,21 @@ Halconfig parseConfig(InputStream is) throws UnrecognizedPropertyException {
    * @return The fully parsed halconfig.
    * @throws UnrecognizedPropertyException
    */
-  public Halconfig getConfig() throws UnrecognizedPropertyException {
+  public Halconfig getConfig() {
     Halconfig res = null;
     try {
       InputStream is = new FileInputStream(new File(halconfigPath));
       res = parseConfig(is);
     } catch (FileNotFoundException e) {
-      res = new Halconfig().setHalyardVersion(halyardVersion);
+      throw new NoConfigException(halconfigPath);
+    } catch (UnrecognizedPropertyException e) {
+      throw new ParseConfigException(e);
+    } catch (ParserException e) {
+      throw new ParseConfigException(e);
+    } catch (ScannerException e) {
+      throw new ParseConfigException(e);
```

Comment:
If I use that, `ParseConfigException` is handed an `Exception`, and I can't delineate between errors in my Constructor without inspecting the class type, which feels messy.